### PR TITLE
FEAT: implemented add to Wishlist

### PR DESCRIPTION
**Pull request by shivanigangadharan**
Implemented the following features :- 

- Added product to wishlist (heart button on top right of card)
- Linked cart/wishlist items number in navbar 
- Added view products button if wishlist/cart is empty.

### CODE REVIEW

File: src/components/cartCard/cartCard.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import React, { useEffect, useState } from 'react';
+import jacket from '../../assets/jacket.jpg';
+import { useAuth } from '../../context/authContext';
+import axios from 'axios';
+
+export default function CartCard({ cartitem }) {
+    const { title, author, brand, imgurl, price, rating, _id, id } = cartitem;
+    // console.log("cart", cartitem);
```
Comment (abhijeetekad) [inline src/components/cartCard/cartCard.js]: Remove console.log()

File: src/context/authContext.js [hunk 2]
```diff
@@ -1,38 +1,38 @@
...
             console.log("Found error: ", e);
-            setError({
-                errorExists: true,
-                errorMessage: e
-            })
+            // setError({
+            //     errorExists: true,
+            //     errorMessage: e
+            // })
+            alert("Credentials invalid. Please sign-up if you don't already have an account.");
```
Comment (abhijeetekad) [inline src/context/authContext.js]: Commented code can be removed if not required anymore.

**Review by abhijeetekad — APPROVED** [review]
code looks great. You can merge.

File: src/components/cartCard/cartCard.js (hunk 1 above)
Comment (shivanigangadharan) [reply src/components/cartCard/cartCard.js]: done

File: src/context/authContext.js (hunk 2 above)
Comment (shivanigangadharan) [reply src/context/authContext.js]: done

**Review by AmishMishra11 — COMMENTED** [review]
The login feature is not working Please fix that

In the cart please change the heart color if it's not in the wishlist and change it to red when item is added to the wishlist

add remove from wishlist button

same items are being added to wishlist multiple times so please look into it

**Comment by shivanigangadharan** [comment]
> The login feature is not working Please fix that
> 
> In the cart please change the heart color if it's not in the wishlist and change it to red when item is added to the wishlist
> 
> add remove from wishlist button
> 
> same items are being added to wishlist multiple times so please look into it

Made all of these changes in the next branch, merging this one.

**Pull request merged**